src/main/java/org/apache/sysml/runtime/controlprogram/ProgramBlock.java

```diff
@@ -363,7 +368,16 @@ private static void checkSparsity( Instruction lastInst, LocalVariableMap vars )
 					synchronized( mb ) { //potential state change
 						mb.recomputeNonZeros();
 						mb.examSparsity();
+
 					}
+					if(sparse1) {
+						int rlen = mb.getNumRows();
+						int clen = mb.getNumColumns();
+						long nnz = mb.getNonZeros();
+						sparseBlock = mb.getSparseBlock();
+						sparseBlock.checkValidity(rlen, clen, nnz, true);
```

Comment:
As note above, please don't assign the sparse block to a member variable and call mb.getSparseBlock().checkValidity() instead.